vscode-extension/src/extension.ts

```diff
@@ -1,141 +1,23 @@
-    );
-
-    return;
-  }
-
-  outputChannel.appendLine(`Using relay binary: ${relayBinary}`);
-
-  const serverOptions: ServerOptions = {
-    command: relayBinary,
-    args: ['lsp', `--output=${outputLevel}`],
-  };
-
-  // Options to control the language client
-  const clientOptions: LanguageClientOptions = {
-    markdown: {
-      isTrusted: true,
-    },
-    documentSelector: [
-      {scheme: 'file', language: 'javascript'},
-      {scheme: 'file', language: 'typescript'},
-      {scheme: 'file', language: 'typescriptreact'},
-      {scheme: 'file', language: 'javascriptreact'},
-    ],
-
-    outputChannel,
-
-    // Since we use stderr for debug logs, the "Something went wrong" popup
-    // in VSCode shows up a lot. This tells vscode not to show it in any case.
-    revealOutputChannelOn: RevealOutputChannelOn.Never,
-
-    initializationFailedHandler: (error) => {
-      outputChannel.appendLine(`initializationFailedHandler ${error}`);
-
-      return true;
-    },
-
-    errorHandler: {
-      // This happens when the LSP server stops running.
-      // e.g. Could not find relay config.
-      // e.g. watchman was not installed.
-      //
-      // TODO: Figure out the best way to handle this `closed` event
-      //
-      // Some of these messages are worth surfacing and others are not
-      // e.g. "Watchman is not installed" is important to surface to the user
-      // but "No relay config found" is not relevant since the user is likely
-      // just in a workspace where they don't have a relay config.
-      //
-      // We already bail early if there is no relay binary found.
-      // So maybe we should just show all of these messages since it would
-      // be weird if you had a relay binary in your node modules but no relay
-      // config could be found. 🤷 for now.
-      closed() {
-        window
-          .showWarningMessage(
-            'Relay LSP client connection got closed unexpectedly.',
-            'Go to output',
-            'Ignore',
-          )
-          .then((selected) => {
-            if (selected === 'Go to output') {
-              client.outputChannel.show();
-            }
-          });
-
-        return CloseAction.DoNotRestart;
-      },
-      // This `error` callback should probably never happen. 🙏
-      error() {
-        window
-          .showWarningMessage(
-            'An error occurred while writing/reading to/from the relay lsp connection',
-            'Go to output',
-            'Ignore',
-          )
-          .then((selected) => {
-            if (selected === 'Go to output') {
-              client.outputChannel.show();
-            }
-          });
-
-        return ErrorAction.Continue;
-      },
-    },
+export async function activate(extensionContext: ExtensionContext) {
+  relayExtensionContext = {
+    client: null,
+    outputChannel: window.createOutputChannel('Relay Language Server'),
```

Comment:
Do we need to ensure that the output channel gets its `.dispose` method called when the extension is closed?

Maybe add it to the context's subscriptions?